linera-views/src/backends/mod.rs

```diff
@@ -15,13 +15,180 @@ pub mod lru_caching;
+    crate::backends::dynamo_db::DynamoDbStoreInternalError,
+>;
+
+/// The config type for DynamoDbStore
+#[cfg(with_dynamodb)]
+pub type DynamoDbStoreConfig =
+    crate::lru_caching::LruSplittingConfig<crate::backends::dynamo_db::DynamoDbStoreInternalConfig>;
+
+#[cfg(with_dynamodb)]
+impl DynamoDbStoreConfig {
+    /// Creates a `DynamoDbStoreConfig` from the input.
+    pub fn new(
+        config: aws_sdk_dynamodb::Config,
+        common_config: crate::store::CommonStoreConfig,
+    ) -> DynamoDbStoreConfig {
+        let inner_config = crate::backends::dynamo_db::DynamoDbStoreInternalConfig {
+            config,
+            common_config: common_config.reduced(),
+        };
+        DynamoDbStoreConfig {
+            inner_config,
+            cache_size: common_config.cache_size,
+        }
+    }
+}
+
+/// Getting a configuration for the system
+#[cfg(with_dynamodb)]
+pub async fn get_config(
+    use_localstack: bool,
+) -> Result<aws_sdk_dynamodb::Config, DynamoDbStoreError> {
+    Ok(crate::backends::dynamo_db::get_config_internal(use_localstack).await?)
+}
+
+/// The `ScyllaDbStore` composed type with metrics
+#[cfg(all(with_scylladb, with_metrics))]
+pub type ScyllaDbStore = crate::metering::MeteredStore<
+    crate::lru_caching::LruCachingStore<
+        crate::metering::MeteredStore<
+            crate::value_splitting::ValueSplittingStore<
+                crate::metering::MeteredStore<
+                    crate::journaling::JournalingKeyValueStore<
+                        crate::backends::scylla_db::ScyllaDbStoreInternal,
+                    >,
+                >,
+            >,
+        >,
+    >,
+>;
+
+/// The `ScyllaDbStore` composed type
+#[cfg(all(with_scylladb, not(with_metrics)))]
+pub type ScyllaDbStore = crate::lru_caching::LruCachingStore<
+    crate::value_splitting::ValueSplittingStore<
+        crate::journaling::JournalingKeyValueStore<
+            crate::backends::scylla_db::ScyllaDbStoreInternal,
+        >,
+    >,
+>;
+
+/// The `ScyllaDbStoreConfig` input type
+#[cfg(with_scylladb)]
+pub type ScyllaDbStoreConfig =
+    crate::lru_caching::LruSplittingConfig<crate::backends::scylla_db::ScyllaDbStoreInternalConfig>;
+
+#[cfg(with_scylladb)]
+impl ScyllaDbStoreConfig {
+    /// Creates a `ScyllaDbStoreConfig` from the inputs.
+    pub fn new(uri: String, common_config: crate::store::CommonStoreConfig) -> ScyllaDbStoreConfig {
+        let inner_config = crate::backends::scylla_db::ScyllaDbStoreInternalConfig {
+            uri,
+            common_config: common_config.reduced(),
+        };
+        ScyllaDbStoreConfig {
+            inner_config,
+            cache_size: common_config.cache_size,
+        }
+    }
+}
+
+/// The combined error type for the `ScyllaDbStore`.
+#[cfg(with_scylladb)]
+pub type ScyllaDbStoreError = crate::value_splitting::ValueSplittingError<
+    crate::backends::scylla_db::ScyllaDbStoreInternalError,
+>;
```

Comment:
We could still have these in separate submodules and re-export them, to reduce the number of `cfg` statements.

In general I'm not sure if moving these directly into `backends` is an improvement.